File: frontend/src/app/models/Book.ts
export interface BookFull extends Book {
  description?: string
  price?: number
  pages?: number
  year?: number
  categories?: Set<Category>
  publisher?: Publisher
  genre?: Genre
  series?: Series
  reviews?: Set<Review>
}

export interface Book {
  title: string
  estimation?: number
  isbn: string
  authors?: Set<Author>
  cover: Cover
}

export interface Cover {
  path: string
  extension: string
}

export interface Author {
  name: string
  role: string
}

export interface Category {
  title: string
}

export interface Publisher {
  title: string
  description: string
}

export interface Genre {
  path: string
}

export interface Series {
  title: string
  description: string
}

export interface Review {
  //todo add title and date
  //title?: string
  review: string
  estimation: number
  book: BookPk
  profileDto?: ProfileDto
  //date?: Date
}

export interface BookPk {
  isbn: string
}

export interface ProfileDto {
  id: number;
  firstName: string
  lastName: string
}

File: backend/html-parsers/labirint/src/main/java/ru/bookfind/parsers/labirint/dtos/Book.java
package ru.bookfind.parsers.labirint.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;

import java.util.HashSet;
import java.util.Set;

@Data
public class Book {

    private Long labirintId;

    private String title;

    private String description;

    private Integer price;

    private Integer pages;

    private Integer year;

    private Float estimation;

    private Integer estimationsCount;

    private String isbn;

    private Set<Isbn> isbns = new HashSet<>();

    private Status status;

    private Set<Category> categories;

    private Set<Author> authors;

    private Publisher publisher;

    private Genre genre;

    private Series series;

    private Cover cover;

    @AllArgsConstructor
    @Getter
    public enum Status {
        ACTIVE, HIDDEN;
    }
}